client/my-sites/patterns/components/pattern-library/index.tsx

```diff
@@ -103,17 +104,18 @@ export const PatternLibrary = ( {
 	const isLoggedIn = useSelector( isUserLoggedIn );
 	const isDevAccount = useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
 
-	const handleViewChange = ( view: 'grid' | 'list' ) => {
-		const currentView = isGridView ? 'grid' : 'list';
+	const type = patternTypeFilter === PatternTypeFilter.REGULAR ? 'pattern' : 'page-layout';
+	const currentView = isGridView ? 'grid' : 'list';
 
+	const handleViewChange = ( view: PatternView ) => {
 		if ( currentView === view ) {
 			return;
 		}
 
 		recordTracksEvent( 'calypso_pattern_library_view_switch', {
 			category,
 			is_logged_in: isLoggedIn,
-			type: patternTypeFilter === PatternTypeFilter.REGULAR ? 'pattern' : 'page-layout',
+			type,
 			user_is_dev_account: isDevAccount ? '1' : '0',
 			view,
 		} );
```

Comment:
Maybe we should consider moving this into `PatternsPageViewTracker` now that we pass the `view` prop to that component?